In [1]:
# Loading libraries and files
import numpy as np
import pandas as pd
# pd.set_option('display.precision',3)
pd.set_option('display.float_format', lambda x: '%.2f' % x)

companies = pd.read_csv("companies.txt", sep="\t", encoding = "ISO-8859-1")
rounds2 = pd.read_csv("rounds2.csv", sep=",", encoding = "ISO-8859-1")

In [2]:
# Fixing non-English characters

companies.permalink = companies.permalink.str.encode("ISO-8859-1").str.decode("ascii", "ignore")
rounds2.company_permalink = rounds2.company_permalink.str.encode("ISO-8859-1").str.decode("ascii", "ignore")

In [3]:
# Finding the number of unique companies in round2

rounds2["company_permalink"] = rounds2["company_permalink"].str.lower()
len(rounds2["company_permalink"].str.lower().unique().tolist())

66368

In [4]:
# Finding the number of unique companies in round2

companies["permalink"] = companies["permalink"].str.lower()
len(companies["permalink"].str.lower().unique().tolist())

66368

In [5]:
# To check whether all companies in rounds2 are present in companies

len(set(companies["permalink"]).intersection(set(rounds2["company_permalink"])))

66368

In [6]:
# Merging rounds2 and companies

companies.rename(columns={"permalink":"company_permalink"}, inplace=True)
master_frame = pd.merge(rounds2, companies, how="inner", on="company_permalink")

In [7]:
# shape of the master_frame

master_frame.shape

(114949, 15)

In [8]:
master_frame.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 114949 entries, 0 to 114948
Data columns (total 15 columns):
company_permalink          114949 non-null object
funding_round_permalink    114949 non-null object
funding_round_type         114949 non-null object
funding_round_code         31140 non-null object
funded_at                  114949 non-null object
raised_amount_usd          94959 non-null float64
name                       114948 non-null object
homepage_url               108815 non-null object
category_list              111539 non-null object
status                     114949 non-null object
country_code               106271 non-null object
state_code                 104003 non-null object
region                     104782 non-null object
city                       104785 non-null object
founded_at                 94428 non-null object
dtypes: float64(1), object(14)
memory usage: 14.0+ MB


In [9]:
# Analysing the columns that have missing data
master_frame.isnull().any()

company_permalink          False
funding_round_permalink    False
funding_round_type         False
funding_round_code          True
funded_at                  False
raised_amount_usd           True
name                        True
homepage_url                True
category_list               True
status                     False
country_code                True
state_code                  True
region                      True
city                        True
founded_at                  True
dtype: bool

In [10]:
# Sum of missing values column-wise
master_frame.isnull().sum()

company_permalink              0
funding_round_permalink        0
funding_round_type             0
funding_round_code         83809
funded_at                      0
raised_amount_usd          19990
name                           1
homepage_url                6134
category_list               3410
status                         0
country_code                8678
state_code                 10946
region                     10167
city                       10164
founded_at                 20521
dtype: int64

In [11]:
# Missing data in terms of percentage
round(100*(master_frame.isnull().sum()/len(master_frame.index)), 2)

company_permalink          0.00
funding_round_permalink    0.00
funding_round_type         0.00
funding_round_code        72.91
funded_at                  0.00
raised_amount_usd         17.39
name                       0.00
homepage_url               5.34
category_list              2.97
status                     0.00
country_code               7.55
state_code                 9.52
region                     8.84
city                       8.84
founded_at                17.85
dtype: float64

In [12]:
master_frame

company_permalink  \
0                                     /organization/-fame   
1                                  /organization/-qounter   
2                                  /organization/-qounter   
3                     /organization/-the-one-of-them-inc-   
4                                   /organization/0-6-com   
5                          /organization/004-technologies   
6                        /organization/01games-technology   
7                     /organization/0ndine-biomedical-inc   
8                     /organization/0ndine-biomedical-inc   
9                                    /organization/0xdata   
10                                   /organization/0xdata   
11                                   /organization/0xdata   
12                                   /organization/0xdata   
13                                        /organization/1   
14                                        /organization/1   
15                                        /organization/1   
16                              /organization/1-2-3-listo   
17                                  /organization/1-4-all   
18                         /organization/1-618-technology   
19                            /organization/1-800-dentist   
20                            /organization/1-800-doctors   
21               /organization/1-800-publicrelations-inc-   
22                             /organization/1-mainstream   
23                                  /organization/1-of-99   
24                              /organization/10-20-media   
25                              /organization/10-20-media   
26                              /organization/10-20-media   
27                              /organization/10-20-media   
28                          /organization/10-minutes-with   
29                          /organization/10-minutes-with   
...                                                   ...   
114919                              /organization/zynstra   
114920                           /organization/zyomyx-inc   
114921                           /organization/zyomyx-inc   
114922                           /organization/zyomyx-inc   
114923                           /organization/zyomyx-inc   
114924                                 /organization/zype   
114925                                 /organization/zype   
114926                               /organization/zypsee   
114927                               /organization/zypsee   
114928                     /organization/zyraz-technology   
114929                     /organization/zyraz-technology   
114930                     /organization/zyraz-technology   
114931                     /organization/zyraz-technology   
114932                                /organization/zyrra   
114933                                /organization/zyrra   
114934                                /organization/zyrra   
114935                                /organization/zyrra   
114936                               /organization/zystor   
114937                           /organization/zytoprotec   
114938                                /organization/zzish   
114939                                /organization/zzish   
114940                                /organization/zzish   
114941  /organization/zznode-science-and-technology-co...   
114942                          /organization/zzzzapp-com   
114943                          /organization/zzzzapp-com   
114944                          /organization/zzzzapp-com   
114945                          /organization/zzzzapp-com   
114946                                 /organization/eron   
114947                               /organization/asys-2   
114948  /organization/novatiff-reklam-ve-tantm-hizmetl...   

                                funding_round_permalink   funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638              venture   
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633              venture   
2       /funding-

In [13]:
# Removing non-contributing column
master_frame = master_frame.drop('funding_round_code', axis=1)
master_frame = master_frame.drop('founded_at', axis=1)

# Views missing data report post initial cleanup 
round(100*(master_frame.isnull().sum()/len(master_frame.index)), 2)

company_permalink          0.00
funding_round_permalink    0.00
funding_round_type         0.00
funded_at                  0.00
raised_amount_usd         17.39
name                       0.00
homepage_url               5.34
category_list              2.97
status                     0.00
country_code               7.55
state_code                 9.52
region                     8.84
city                       8.84
dtype: float64

In [14]:
# Removing rows where raised_amount_usd is null
master_frame = master_frame[~np.isnan(master_frame['raised_amount_usd'])]

# Missing data report post row deletion 
round(100*(master_frame.isnull().sum()/len(master_frame.index)), 2)

company_permalink         0.00
funding_round_permalink   0.00
funding_round_type        0.00
funded_at                 0.00
raised_amount_usd         0.00
name                      0.00
homepage_url              4.56
category_list             1.10
status                    0.00
country_code              6.16
state_code                8.01
region                    7.42
city                      7.42
dtype: float64

In [15]:
# Number of rows where all values are missing

len(master_frame[master_frame.isnull().all(axis=1)==True].index)

0

In [16]:
# Removing unnecessary columns
master_frame = master_frame.drop('homepage_url', axis=1)
master_frame = master_frame.drop('state_code', axis=1)
master_frame = master_frame.drop('region', axis=1)
master_frame = master_frame.drop('city', axis=1)

# Percentage of missing data still present
round(100*(master_frame.isnull().sum()/len(master_frame.index)), 2)

company_permalink         0.00
funding_round_permalink   0.00
funding_round_type        0.00
funded_at                 0.00
raised_amount_usd         0.00
name                      0.00
category_list             1.10
status                    0.00
country_code              6.16
dtype: float64

In [17]:
# Percentage of original data after cleanup
len(master_frame.index)/114949

0.8260967907506808

We now have 82.6% of the original data and have lost about 17.4% of the original data.

In [18]:
pd.options.mode.chained_assignment = None

# Imputing NaNs by UNK for country_code
master_frame.loc[pd.isnull(master_frame['country_code']), ['country_code']] = 'UNK'

# Imputing NaNs by undefined for category_list
master_frame.loc[pd.isnull(master_frame['category_list']), ['category_list']] = "undefined"
round(100*(master_frame.isnull().sum()/len(master_frame.index)), 2)

company_permalink         0.00
funding_round_permalink   0.00
funding_round_type        0.00
funded_at                 0.00
raised_amount_usd         0.00
name                      0.00
category_list             0.00
status                    0.00
country_code              0.00
dtype: float64

In [19]:
# Number of observations in master_frame

len(master_frame.index)

94959

In [20]:
master_frame
#exporting master frame
master_frame.to_csv('F://IIITB//Module//Project//Latest//master.csv', sep=',')

In [21]:
master_frame.groupby("funding_round_type")['raised_amount_usd'].mean()

funding_round_type
angel                      958694.47
convertible_note          1453438.54
debt_financing           17043526.02
equity_crowdfunding        538368.21
grant                     4300576.34
non_equity_assistance      411203.05
post_ipo_debt           168704571.82
post_ipo_equity          82182493.87
private_equity           73308593.03
product_crowdfunding      1363131.07
secondary_market         79649630.10
seed                       719818.00
undisclosed              19242370.23
venture                  11748949.13
Name: raised_amount_usd, dtype: float64

In [22]:
# The investment type that is most suitable for Spark Funds

(master_frame.groupby("funding_round_type").mean() >= 5000000) & (master_frame.groupby("funding_round_type").mean() <= 15000000)

raised_amount_usd
funding_round_type                      
angel                              False
convertible_note                   False
debt_financing                     False
equity_crowdfunding                False
grant                              False
non_equity_assistance              False
post_ipo_debt                      False
post_ipo_equity                    False
private_equity                     False
product_crowdfunding               False
secondary_market                   False
seed                               False
undisclosed                        False
venture                             True

In [23]:
master_frame = master_frame[master_frame["funding_round_type"]=="venture"]
master_frame

company_permalink  \
0                                     /organization/-fame   
3                     /organization/-the-one-of-them-inc-   
4                                   /organization/0-6-com   
8                     /organization/0ndine-biomedical-inc   
10                                   /organization/0xdata   
11                                   /organization/0xdata   
12                                   /organization/0xdata   
22                             /organization/1-mainstream   
28                          /organization/10-minutes-with   
34                             /organization/1000memories   
38                          /organization/1000museums-com   
39                          /organization/1000museums-com   
41                          /organization/1000museums-com   
44                          /organization/1000museums-com   
46                               /organization/1001-menus   
47                               /organization/1001-menus   
49                               /organization/1001-menus   
50                       /organization/1001pharmacies-com   
52                                  /organization/1006-tv   
55                                 /organization/100du-tv   
56                                 /organization/100e-com   
57                                 /organization/100e-com   
59                                /organization/100health   
61                                  /organization/100plus   
62                                 /organization/1010data   
73                             /organization/10x-genomics   
74                         /organization/10x-technologies   
80                                    /organization/117go   
81                            /organization/11i-solutions   
95                                /organization/1234enter   
...                                                   ...   
114886                              /organization/zymetis   
114889                            /organization/zymeworks   
114890                            /organization/zymeworks   
114891                            /organization/zymeworks   
114892                            /organization/zymeworks   
114893                            /organization/zymeworks   
114894                            /organization/zymeworks   
114900                                /organization/zyncd   
114902                               /organization/zyncro   
114903                               /organization/zyncro   
114904              /organization/zynerba-pharmaceuticals   
114906                                /organization/zynga   
114907                                /organization/zynga   
114908                                /organization/zynga   
114909                                /organization/zynga   
114910                                /organization/zynga   
114911                                /organization/zynga   
114912                                /organization/zynga   
114914                                /organization/zynga   
114915                             /organization/zyngenia   
114916                              /organization/zynstra   
114917                              /organization/zynstra   
114920                           /organization/zyomyx-inc   
114922                           /organization/zyomyx-inc   
114923                           /organization/zyomyx-inc   
114931                     /organization/zyraz-technology   
114934                                /organization/zyrra   
114936                               /organization/zystor   
114937                           /organization/zytoprotec   
114941  /organization/zznode-science-and-technology-co...   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
3       /funding-round/650b8f704416801069bb178a1418776b            venture   
4       /funding-round/

In [24]:
# Extracting the primary sector of each category list from category_list column

# Function to extract only the primary category
def extract_primary_category(value):
    values = value.split('|')
    if len(values) > 1:
        return values[0]
    else:
        return value

master_frame['primary_sector'] = master_frame['category_list'].apply(extract_primary_category)
master_frame['primary_sector'] = master_frame['primary_sector'].str.lower()

master_frame

company_permalink  \
0                                     /organization/-fame   
3                     /organization/-the-one-of-them-inc-   
4                                   /organization/0-6-com   
8                     /organization/0ndine-biomedical-inc   
10                                   /organization/0xdata   
11                                   /organization/0xdata   
12                                   /organization/0xdata   
22                             /organization/1-mainstream   
28                          /organization/10-minutes-with   
34                             /organization/1000memories   
38                          /organization/1000museums-com   
39                          /organization/1000museums-com   
41                          /organization/1000museums-com   
44                          /organization/1000museums-com   
46                               /organization/1001-menus   
47                               /organization/1001-menus   
49                               /organization/1001-menus   
50                       /organization/1001pharmacies-com   
52                                  /organization/1006-tv   
55                                 /organization/100du-tv   
56                                 /organization/100e-com   
57                                 /organization/100e-com   
59                                /organization/100health   
61                                  /organization/100plus   
62                                 /organization/1010data   
73                             /organization/10x-genomics   
74                         /organization/10x-technologies   
80                                    /organization/117go   
81                            /organization/11i-solutions   
95                                /organization/1234enter   
...                                                   ...   
114886                              /organization/zymetis   
114889                            /organization/zymeworks   
114890                            /organization/zymeworks   
114891                            /organization/zymeworks   
114892                            /organization/zymeworks   
114893                            /organization/zymeworks   
114894                            /organization/zymeworks   
114900                                /organization/zyncd   
114902                               /organization/zyncro   
114903                               /organization/zyncro   
114904              /organization/zynerba-pharmaceuticals   
114906                                /organization/zynga   
114907                                /organization/zynga   
114908                                /organization/zynga   
114909                                /organization/zynga   
114910                                /organization/zynga   
114911                                /organization/zynga   
114912                                /organization/zynga   
114914                                /organization/zynga   
114915                             /organization/zyngenia   
114916                              /organization/zynstra   
114917                              /organization/zynstra   
114920                           /organization/zyomyx-inc   
114922                           /organization/zyomyx-inc   
114923                           /organization/zyomyx-inc   
114931                     /organization/zyraz-technology   
114934                                /organization/zyrra   
114936                               /organization/zystor   
114937                           /organization/zytoprotec   
114941  /organization/zznode-science-and-technology-co...   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
3       /funding-round/650b8f704416801069bb178a1418776b            venture   
4       /funding-round/

In [25]:
# Top nine countries which have received the highest total funding for Venture type

master_frame.groupby('country_code')['raised_amount_usd'].sum().sort_values(ascending=False).head(9)

country_code
USA   422510842796.00
CHN    39835418773.00
UNK    25506108414.00
GBR    20245627416.00
IND    14391858718.00
CAN     9583332317.00
FRA     7259536732.00
ISR     6907514579.00
DEU     6346959822.00
Name: raised_amount_usd, dtype: float64

In [26]:
# Creating the dataframe top9
top9 = master_frame
top9.groupby('country_code')['raised_amount_usd'].sum().sort_values(ascending=False)

country_code
USA   422510842796.00
CHN    39835418773.00
UNK    25506108414.00
GBR    20245627416.00
IND    14391858718.00
CAN     9583332317.00
FRA     7259536732.00
ISR     6907514579.00
DEU     6346959822.00
JPN     3363676611.00
SWE     3254952563.00
NLD     2939403619.00
CHE     2827560264.00
SGP     2793917856.00
ESP     1835831452.00
BRA     1785986890.00
IRL     1676131350.00
RUS     1570426005.00
AUS     1322934948.00
DNK     1228310641.00
BEL     1070542301.00
FIN     1043199709.00
NOR      956192413.00
KOR      939988262.00
MYS      883058787.00
HKG      781266982.00
AUT      630954952.00
TWN      623979485.00
TUR      559097457.00
ITA      488289353.00
            ...      
KWT       14000000.00
LIE       13091716.00
MNE       12200000.00
SVN       12017508.00
BGR       11300000.00
KAZ       11000000.00
GRC       10743775.00
BAH        8900000.00
TTO        8500000.00
SVK        8241062.00
BGD        7002000.00
LBN        6455000.00
GGY        3960000.00
TUN        3920000.

In [27]:
top9

company_permalink  \
0                                     /organization/-fame   
3                     /organization/-the-one-of-them-inc-   
4                                   /organization/0-6-com   
8                     /organization/0ndine-biomedical-inc   
10                                   /organization/0xdata   
11                                   /organization/0xdata   
12                                   /organization/0xdata   
22                             /organization/1-mainstream   
28                          /organization/10-minutes-with   
34                             /organization/1000memories   
38                          /organization/1000museums-com   
39                          /organization/1000museums-com   
41                          /organization/1000museums-com   
44                          /organization/1000museums-com   
46                               /organization/1001-menus   
47                               /organization/1001-menus   
49                               /organization/1001-menus   
50                       /organization/1001pharmacies-com   
52                                  /organization/1006-tv   
55                                 /organization/100du-tv   
56                                 /organization/100e-com   
57                                 /organization/100e-com   
59                                /organization/100health   
61                                  /organization/100plus   
62                                 /organization/1010data   
73                             /organization/10x-genomics   
74                         /organization/10x-technologies   
80                                    /organization/117go   
81                            /organization/11i-solutions   
95                                /organization/1234enter   
...                                                   ...   
114886                              /organization/zymetis   
114889                            /organization/zymeworks   
114890                            /organization/zymeworks   
114891                            /organization/zymeworks   
114892                            /organization/zymeworks   
114893                            /organization/zymeworks   
114894                            /organization/zymeworks   
114900                                /organization/zyncd   
114902                               /organization/zyncro   
114903                               /organization/zyncro   
114904              /organization/zynerba-pharmaceuticals   
114906                                /organization/zynga   
114907                                /organization/zynga   
114908                                /organization/zynga   
114909                                /organization/zynga   
114910                                /organization/zynga   
114911                                /organization/zynga   
114912                                /organization/zynga   
114914                                /organization/zynga   
114915                             /organization/zyngenia   
114916                              /organization/zynstra   
114917                              /organization/zynstra   
114920                           /organization/zyomyx-inc   
114922                           /organization/zyomyx-inc   
114923                           /organization/zyomyx-inc   
114931                     /organization/zyraz-technology   
114934                                /organization/zyrra   
114936                               /organization/zystor   
114937                           /organization/zytoprotec   
114941  /organization/zznode-science-and-technology-co...   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
3       /funding-round/650b8f704416801069bb178a1418776b            venture   
4       /funding-round/

In [28]:
top9_english=top9.groupby('country_code')['raised_amount_usd'].sum().sort_values(ascending=False)
top9_english

country_code
USA   422510842796.00
CHN    39835418773.00
UNK    25506108414.00
GBR    20245627416.00
IND    14391858718.00
CAN     9583332317.00
FRA     7259536732.00
ISR     6907514579.00
DEU     6346959822.00
JPN     3363676611.00
SWE     3254952563.00
NLD     2939403619.00
CHE     2827560264.00
SGP     2793917856.00
ESP     1835831452.00
BRA     1785986890.00
IRL     1676131350.00
RUS     1570426005.00
AUS     1322934948.00
DNK     1228310641.00
BEL     1070542301.00
FIN     1043199709.00
NOR      956192413.00
KOR      939988262.00
MYS      883058787.00
HKG      781266982.00
AUT      630954952.00
TWN      623979485.00
TUR      559097457.00
ITA      488289353.00
            ...      
KWT       14000000.00
LIE       13091716.00
MNE       12200000.00
SVN       12017508.00
BGR       11300000.00
KAZ       11000000.00
GRC       10743775.00
BAH        8900000.00
TTO        8500000.00
SVK        8241062.00
BGD        7002000.00
LBN        6455000.00
GGY        3960000.00
TUN        3920000.

From the above list, the english speaking countries are USA, GBR, IND, CAN

In [29]:
# Top 3 English speaking countries

top3_english = top9[top9["country_code"].isin(['USA', 'GBR', 'IND'])]
top3_english.groupby('country_code')['raised_amount_usd'].sum().sort_values(ascending=False)

country_code
USA   422510842796.00
GBR    20245627416.00
IND    14391858718.00
Name: raised_amount_usd, dtype: float64

In [30]:
# Top 9 english speaking countries.
top9_english_conuntries = top9[top9["country_code"].isin(['USA', 'GBR', 'IND','CAN','IRL','AUS','NZL','NGA','ZAF'])]

In [31]:
#exporting top9_english_conuntries
top9_english_conuntries.to_csv('F://IIITB//Module//Project//Latest//top9_english_conuntries.csv', sep=',')

In [32]:
# Loading mapping csv into dataframe
mapping = pd.read_csv("mapping.csv", encoding = "ISO-8859-1")
mapping

category_list  Automotive & Sports  Blanks  \
0                           NaN                    0       1   
1                            3D                    0       0   
2                   3D Printing                    0       0   
3                 3D Technology                    0       0   
4                    Accounting                    0       0   
5              Active Lifestyle                    0       0   
6                  Ad Targeting                    0       0   
7            Advanced Materials                    0       0   
8              Adventure Travel                    1       0   
9                   Advertising                    0       0   
10        Advertising Exchanges                    0       0   
11         Advertising Networks                    0       0   
12        Advertising Platforms                    0       0   
13                       Advice                    0       0   
14                    Aerospace                    1       0   
15                  Agriculture                    0       0   
16        Air Pollution Control                    0       0   
17                   Algorithms                    0       0   
18                  All Markets                    0       0   
19                 All Students                    0       0   
20          Alter0tive Medicine                    0       0   
21                       Alumni                    0       0   
22                     A0lytics                    0       0   
23                      Android                    0       0   
24                       Angels                    0       0   
25                  Animal Feed                    0       0   
26   Anything Capital Intensive                    0       0   
27                App Discovery                    0       0   
28                App Marketing                    0       0   
29                   App Stores                    0       0   
..                          ...                  ...     ...   
658                   Usability                    0       0   
659      User Experience Design                    0       0   
660              User Interface                    0       0   
661                   Utilities                    0       0   
662     Vending and Concessions                    0       0   
663             Venture Capital                    0       0   
664                   Veteri0ry                    0       0   
665                       Video                    0       0   
666          Video Conferencing                    0       0   
667                 Video Games                    0       0   
668             Video on Demand                    0       0   
669             Video Streaming                    0       0   
670          Virtual Workforces                    0       0   
671                        VoIP                    0       0   
672             Waste Ma0gement                    0       0   
673                       Watch                    0       0   
674                       Water                    0       0   
675          Water Purification                    0       0   
676                   Wearables                    0       0   
677                Web Browsers                    0       0   
678                  Web Design                    0       0   
679             Web Development                    0       0   
680                 Web Hosting                    0       0   
681                   Web Tools                    0       0   
682                    Weddings                    0       0   
683                   Wholesale                    0       0   
684            Wine And Spirits                    0       0   
685                    Wireless                    0       0   
686                       Women                    0       0   
687                Young Adults                    0       0   

     Cleantech / Semiconductors  Entertainment  H

In [33]:
# Mapping the primary_sector to main_sector

mapping = pd.melt(mapping,id_vars=["category_list"])
mapping = mapping[mapping.value != 0]
mapping = mapping.drop('value', axis=1)
mapping.rename(columns={'variable':'main_sector','category_list':'primary_sector'}, inplace=True)
mapping['primary_sector'] = mapping['primary_sector'].str.lower()
mapping

primary_sector                              main_sector
8             adventure travel                      Automotive & Sports
14                   aerospace                      Automotive & Sports
45                        auto                      Automotive & Sports
46             automated kiosk                      Automotive & Sports
47                  automotive                      Automotive & Sports
57                    bicycles                      Automotive & Sports
69            boating industry                      Automotive & Sports
87                         cad                      Automotive & Sports
93                        cars                      Automotive & Sports
188                     design                      Automotive & Sports
189                  designers                      Automotive & Sports
213                     drones                      Automotive & Sports
226          electric vehicles                      Automotive & Sports
292                      fuels                      Automotive & Sports
331             heavy industry                      Automotive & Sports
364      industrial automation                      Automotive & Sports
409                 limousines                      Automotive & Sports
602                     soccer                      Automotive & Sports
626             sporting goods                      Automotive & Sports
627                     sports                      Automotive & Sports
652             transportation                      Automotive & Sports
657       unmanned air systems                      Automotive & Sports
688                        NaN                                   Blanks
1392     air pollution control               Cleantech / Semiconductors
1409               aquaculture               Cleantech / Semiconductors
1437                 bio-pharm               Cleantech / Semiconductors
1438                  biofuels               Cleantech / Semiconductors
1439            bioinformatics               Cleantech / Semiconductors
1440  biomass power generation               Cleantech / Semiconductors
1441                biometrics               Cleantech / Semiconductors
...                        ...                                      ...
6020            perso0l fi0nce  Social, Finance, Analytics, Advertising
6037                  pre seed  Social, Finance, Analytics, Advertising
6038       predictive a0lytics  Social, Finance, Analytics, Advertising
6040          price comparison  Social, Finance, Analytics, Advertising
6053                promotio0l  Social, Finance, Analytics, Advertising
6054        property ma0gement  Social, Finance, Analytics, Advertising
6055          public relations  Social, Finance, Analytics, Advertising
6074                    retail  Social, Finance, Analytics, Advertising
6075         retail technology  Social, Finance, Analytics, Advertising
6079            risk ma0gement  Social, Finance, Analytics, Advertising
6082       sales and marketing  Social, Finance, Analytics, Advertising
6107   social + mobile + local  Social, Finance, Analytics, Advertising
6108        social bookmarking  Social, Finance, Analytics, Advertising
6109           social business  Social, Finance, Analytics, Advertising
6110             social buying  Social, Finance, Analytics, Advertising
6111           social commerce  Social, Finance, Analytics, Advertising
6112                social crm  Social, Finance, Analytics, Advertising
6113        social fundraising  Social, Finance, Analytics, Advertising
6114              social games  Social, Finance, Analytics, Advertising
6115              social media  Social, Finance, Analytics, Advertising
6116    social media ma0gement  Social, Finance, Analytics, Advertising
6117    social media marketing  Social, Finance, Analytics, Advertising
6118    social media platforms  Social, Finance, Analytics, Advertising
6119      social network media  Social, Finance, Analyti

In [34]:
top3_english = pd.merge(top3_english, mapping, how='left', on='primary_sector', indicator=True)
len(top3_english.index)

39018

In [35]:
left_only = top3_english[top3_english['_merge'] != 'both']
left_only.head()

company_permalink  \
1          /organization/0xdata   
2          /organization/0xdata   
3          /organization/0xdata   
12        /organization/100plus   
43  /organization/1world-online   

                            funding_round_permalink funding_round_type  \
1   /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
2   /funding-round/ae2a174c06517c2394aed45006322a7e            venture   
3   /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
12  /funding-round/b5facb0d9dea2f0352b5834892c88c53            venture   
43  /funding-round/32936e588a134502712877150198a0b3            venture   

     funded_at  raised_amount_usd           name  \
1   09-11-2015        20000000.00         H2O.ai   
2   03-01-2013         1700000.00         H2O.ai   
3   19-07-2014         8900000.00         H2O.ai   
12  02-11-2011          500000.00        100Plus   
43  13-08-2015         2500000.00  1World Online   

                                        category_list     status country_code  \
1                                           Analytics  operating          USA   
2                                           Analytics  operating          USA   
3                                           Analytics  operating          USA   
12                                          Analytics   acquired          USA   
43  Analytics|Big Data|Enterprise Software|Market ...  operating          USA   

   primary_sector main_sector     _merge  
1       analytics         NaN  left_only  
2       analytics         NaN  left_only  
3       analytics         NaN  left_only  
12      analytics         NaN  left_only  
43      analytics         NaN  left_only

In [36]:
left_only

company_permalink  \
1                                /organization/0xdata   
2                                /organization/0xdata   
3                                /organization/0xdata   
12                              /organization/100plus   
43                        /organization/1world-online   
44                        /organization/1world-online   
45                        /organization/1world-online   
67                            /organization/24-7-card   
151                         /organization/3pl-central   
214              /organization/60frames-entertainment   
226    /organization/6th-wave-innovations-corporation   
227                    /organization/7-billion-people   
239                          /organization/7park-data   
265                             /organization/9lenses   
266                             /organization/9lenses   
267                             /organization/9lenses   
278                      /organization/a-little-world   
299                         /organization/abacast-inc   
358                            /organization/aboutone   
359                            /organization/aboutone   
369                                /organization/abra   
373                         /organization/absolutdata   
393                          /organization/acadiasoft   
394                          /organization/acadiasoft   
411            /organization/accelerated-vision-group   
412            /organization/accelerated-vision-group   
413                    /organization/accelerator-corp   
414                    /organization/accelerator-corp   
415                    /organization/accelerator-corp   
430                            /organization/accelops   
...                                               ...   
38551                            /organization/zecter   
38552                            /organization/zecter   
38604                        /organization/zenpayroll   
38605                        /organization/zenpayroll   
38618                          /organization/zentrick   
38627                               /organization/zeo   
38628                               /organization/zeo   
38631                     /organization/zephyr-health   
38632                     /organization/zephyr-health   
38633                     /organization/zephyr-health   
38640                            /organization/zeptor   
38666                        /organization/zeros-ones   
38677                       /organization/zestfinance   
38678                       /organization/zestfinance   
38679                       /organization/zestfinance   
38680                       /organization/zestfinance   
38710                           /organization/ziegler   
38795                         /organization/ziprealty   
38855                      /organization/zonarsystems   
38871                          /organization/zoomdata   
38872                          /organization/zoomdata   
38878                          /organization/zoominfo   
38882                         /organization/zoomsafer   
38883                         /organization/zoomsafer   
38888                            /organization/zoopla   
38889                            /organization/zoopla   
38903                              /organization/zopa   
38904                              /organization/zopa   
38905                              /organization/zopa   
38906                              /organization/zopa   

                               funding_round_permalink funding_round_type  \
1      /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
2      /funding-round/ae2a174c06517c2394aed45006322a7e            venture   
3      /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
12     /funding-round/b5facb0d9dea2f0352b5834892c88c53            venture   
43     /funding-round/32936e588a134502712877150198a0b3            venture   
44     /funding-round/4e30bd5c85d816323

In [37]:
# Unique primary sector name in top3_english which don't map to mapping dataframe
left_only.primary_sector.sort_values(ascending=True).unique()

array(['adaptive equipment', 'alternative medicine', 'analytics',
       'big data analytics', 'biotechnology and semiconductor',
       'business analytics', 'cannabis', 'career management',
       'cloud management', 'contact management',
       'digital rights management', 'digital signage',
       'document management', 'educational games', 'energy management',
       'event management', 'finance', 'finance technology',
       'financial exchanges', 'financial services', 'fleet management',
       'governance', 'greentech', 'identity management',
       'innovation management', 'intellectual asset management',
       'investment management', 'it management', 'journalism',
       'knowledge management', 'lead management', 'mobile analytics',
       'nanotechnology', 'natural gas uses',
       'natural language processing', 'navigation', 'personal finance',
       'personalization', 'product search', 'professional networking',
       'professional services', 'project management',
   

In [38]:
mapping.primary_sector.sort_values(ascending=True)

1847                 0notechnology
1848    0tural language processing
1849              0tural resources
4602                     0vigation
3441                            3d
3442                   3d printing
3443                 3d technology
5526                      a0lytics
5508                    accounting
2757              active lifestyle
5510                  ad targeting
3447            advanced materials
8                 adventure travel
5513                   advertising
5514         advertising exchanges
5515          advertising networks
5516         advertising platforms
4829                        advice
14                       aerospace
3455                   agriculture
1392         air pollution control
4833                    algorithms
4834                   all markets
4835                  all students
2772           alter0tive medicine
4837                        alumni
5527                       android
4840                        angels
2777                

We can observe that in some primary sectors, the 'na' is replaced with '0'' 

In [39]:
# Function to replace 0 with na

import re
mapping['primary_sector'] = mapping.primary_sector.apply(lambda x: re.sub('[0]','na',str(x)))

In [40]:
# Again repeating the above steps

top3_english = top9[top9['country_code'].isin(['USA', 'GBR', 'IND'])]
top3_english = pd.merge(top3_english, mapping, how='left', on='primary_sector', indicator=True)
left_only = top3_english[top3_english['_merge'] != 'both']
left_only.primary_sector.sort_values(ascending=True).unique()
top3_english

company_permalink  \
0                        /organization/-fame   
1                       /organization/0xdata   
2                       /organization/0xdata   
3                       /organization/0xdata   
4                 /organization/1-mainstream   
5              /organization/10-minutes-with   
6                 /organization/1000memories   
7              /organization/1000museums-com   
8              /organization/1000museums-com   
9              /organization/1000museums-com   
10             /organization/1000museums-com   
11                   /organization/100health   
12                     /organization/100plus   
13                    /organization/1010data   
14                /organization/10x-genomics   
15            /organization/10x-technologies   
16               /organization/11i-solutions   
17              /organization/128-technology   
18           /organization/1366-technologies   
19           /organization/1366-technologies   
20           /organization/1366-technologies   
21           /organization/1366-technologies   
22           /organization/1366-technologies   
23           /organization/1366-technologies   
24           /organization/1366-technologies   
25           /organization/1366-technologies   
26                   /organization/140-proof   
27                   /organization/140-proof   
28           /organization/16-mile-solutions   
29                 /organization/170-systems   
...                                      ...   
38988          /organization/zyga-technology   
38989          /organization/zyga-technology   
38990          /organization/zyga-technology   
38991          /organization/zyga-technology   
38992         /organization/zygo-corporation   
38993           /organization/zylie-the-bear   
38994           /organization/zylun-staffing   
38995           /organization/zyme-solutions   
38996           /organization/zyme-solutions   
38997           /organization/zyme-solutions   
38998                 /organization/zymergen   
38999                  /organization/zymetis   
39000                    /organization/zyncd   
39001  /organization/zynerba-pharmaceuticals   
39002                    /organization/zynga   
39003                    /organization/zynga   
39004                    /organization/zynga   
39005                    /organization/zynga   
39006                    /organization/zynga   
39007                    /organization/zynga   
39008                    /organization/zynga   
39009                    /organization/zynga   
39010                 /organization/zyngenia   
39011                  /organization/zynstra   
39012                  /organization/zynstra   
39013               /organization/zyomyx-inc   
39014               /organization/zyomyx-inc   
39015               /organization/zyomyx-inc   
39016                    /organization/zyrra   
39017                   /organization/zystor   

                               funding_round_permalink funding_round_type  \
0      /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1      /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
2      /funding-round/ae2a174c06517c2394aed45006322a7e            venture   
3      /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
4      /funding-round/b952cbaf401f310927430c97b68162ea            venture   
5      /funding-round/0faccbbcc5818dc5326469f13f5a8ac8            venture   
6      /funding-round/502bd0e50c27616995e4bdad24605ef8            venture   
7      /funding-round/13be128d655076a025221d7fddc90d68            venture   
8      /funding-round/6aeb32ab3fc05db2b606d7a2466e8c09            venture   
9      /funding-round/89317984885b44f02e4befb9dc2e588c            venture   
10     /funding-round/fd1b3064232ba9b3cfb5653a619e82e8            venture   
11     /funding-round/502f558bee350299fd1ae503e5b9a124            venture   
12     /funding-round/b5facb0d9dea2f0352b5834892c

In [41]:
mapping.primary_sector.sort_values(ascending=True)

3441                                    3d
3442                           3d printing
3443                         3d technology
5508                            accounting
2757                      active lifestyle
5510                          ad targeting
3447                    advanced materials
8                         adventure travel
5513                           advertising
5514                 advertising exchanges
5515                  advertising networks
5516                 advertising platforms
4829                                advice
14                               aerospace
3455                           agriculture
1392                 air pollution control
4833                            algorithms
4834                           all markets
4835                          all students
2772                  alternative medicine
4837                                alumni
5526                             analytics
5527                               android
4840       

Now, the data is more clean

In [42]:
len(left_only)

230

In [43]:
# Removing all the rows from top3_english which still has no mapping to the mapping dataframe
top3_english = top3_english[top3_english['_merge'] == 'both']
top3_english = top3_english.drop('_merge', axis=1)
top3_english

company_permalink  \
0                        /organization/-fame   
1                       /organization/0xdata   
2                       /organization/0xdata   
3                       /organization/0xdata   
4                 /organization/1-mainstream   
5              /organization/10-minutes-with   
6                 /organization/1000memories   
7              /organization/1000museums-com   
8              /organization/1000museums-com   
9              /organization/1000museums-com   
10             /organization/1000museums-com   
11                   /organization/100health   
12                     /organization/100plus   
13                    /organization/1010data   
14                /organization/10x-genomics   
15            /organization/10x-technologies   
16               /organization/11i-solutions   
17              /organization/128-technology   
18           /organization/1366-technologies   
19           /organization/1366-technologies   
20           /organization/1366-technologies   
21           /organization/1366-technologies   
22           /organization/1366-technologies   
23           /organization/1366-technologies   
24           /organization/1366-technologies   
25           /organization/1366-technologies   
26                   /organization/140-proof   
27                   /organization/140-proof   
28           /organization/16-mile-solutions   
29                 /organization/170-systems   
...                                      ...   
38988          /organization/zyga-technology   
38989          /organization/zyga-technology   
38990          /organization/zyga-technology   
38991          /organization/zyga-technology   
38992         /organization/zygo-corporation   
38993           /organization/zylie-the-bear   
38994           /organization/zylun-staffing   
38995           /organization/zyme-solutions   
38996           /organization/zyme-solutions   
38997           /organization/zyme-solutions   
38998                 /organization/zymergen   
38999                  /organization/zymetis   
39000                    /organization/zyncd   
39001  /organization/zynerba-pharmaceuticals   
39002                    /organization/zynga   
39003                    /organization/zynga   
39004                    /organization/zynga   
39005                    /organization/zynga   
39006                    /organization/zynga   
39007                    /organization/zynga   
39008                    /organization/zynga   
39009                    /organization/zynga   
39010                 /organization/zyngenia   
39011                  /organization/zynstra   
39012                  /organization/zynstra   
39013               /organization/zyomyx-inc   
39014               /organization/zyomyx-inc   
39015               /organization/zyomyx-inc   
39016                    /organization/zyrra   
39017                   /organization/zystor   

                               funding_round_permalink funding_round_type  \
0      /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1      /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
2      /funding-round/ae2a174c06517c2394aed45006322a7e            venture   
3      /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   
4      /funding-round/b952cbaf401f310927430c97b68162ea            venture   
5      /funding-round/0faccbbcc5818dc5326469f13f5a8ac8            venture   
6      /funding-round/502bd0e50c27616995e4bdad24605ef8            venture   
7      /funding-round/13be128d655076a025221d7fddc90d68            venture   
8      /funding-round/6aeb32ab3fc05db2b606d7a2466e8c09            venture   
9      /funding-round/89317984885b44f02e4befb9dc2e588c            venture   
10     /funding-round/fd1b3064232ba9b3cfb5653a619e82e8            venture   
11     /funding-round/502f558bee350299fd1ae503e5b9a124            venture   
12     /funding-round/b5facb0d9dea2f0352b5834892c

In [44]:
#exporting top3_english
top3_english.to_csv('F://IIITB//Module//Project//Latest//top3_english_FINAL_MASTER_DATA.csv', sep=',')

In [45]:
# Dropping the rows where investment is not between 5 and 15 million
top3_english = top3_english.drop(top3_english[(top3_english.raised_amount_usd < 5000000)].index)
top3_english = top3_english.drop(top3_english[(top3_english.raised_amount_usd > 15000000)].index)

len(top3_english.index)

d1 = top3_english[top3_english['country_code'] == 'USA']
d2 = top3_english[top3_english['country_code'] == 'GBR']
d3 = top3_english[top3_english['country_code'] == 'IND']

In [95]:
#D1 company received the highest investment in Others and Social, Finance, Analytics, Advertising
d1_others=d1[d1.main_sector == "Others"]
d1_others.groupby(["company_permalink","name"]).agg({"raised_amount_usd" : 'sum'}).sort_values(by="raised_amount_usd",ascending=False)
d1_social=d1[d1.main_sector == "Social, Finance, Analytics, Advertising"]
d1_social.groupby(["company_permalink","name"]).agg({"raised_amount_usd" : 'sum'}).sort_values(by="raised_amount_usd",ascending=False)

raised_amount_usd
company_permalink                       name                                              
/organization/shotspotter               SST Inc. (Formerly ShotSpotter)        67933006.00
/organization/demandbase                Demandbase                             63000000.00
/organization/intacct                   Intacct                                61800000.00
/organization/netbase                   NetBase Solutions                      60600000.00
/organization/lotame                    Lotame                                 59700000.00
/organization/firstrain                 FirstRain                              58344731.00
/organization/choicestream              ChoiceStream                           58300000.00
/organization/optier                    OpTier                                 55600000.00
/organization/damballa                  Damballa                               55000000.00
/organization/zoove                     Zoove                                  54800000.00
/organization/xactly-corp               Xactly Corp                            54500000.00
/organization/hightail                  Hightail                               54000000.00
/organization/chacha                    ChaCha                                 53000000.00
/organization/inrix                     INRIX                                  51100018.00
/organization/visible-measures          Visible Measures                       48500000.00
/organization/univa                     Univa                                  48385708.00
/organization/vuclip                    Vuclip                                 48100000.00
/organization/yume                      YuMe                                   47900000.00
/organization/turn                      Turn                                   45500000.00
/organization/jaspersoft                Jaspersoft                             43500000.00
/organization/edo-interactive           edo Interactive                        43000100.00
/organization/instamed                  InstaMed                               43000000.00
/organization/inkling-systems           Inkling Systems                        42599999.00
/organization/laszlosystems             Laszlo Systems                         41850000.00
/organization/metamarkets               Metamarkets                            41000000.00
/organization/host-analytics            Host Analytics                         40650000.00
/organization/pivotlink-formerly-seatab PivotLink                              40085068.00
/organization/4info                     4INFO                                  40000000.00
/organization/aggregateknowledge        Aggregate Knowledge                    39958350.00
/organization/fiberzone-networks        FiberZone Networks                     39786202.00
...                                                                                    ...
/organization/becovillage               Becovillage                             5000000.00
/organization/sandlinks                 SandLinks                               5000000.00
/organization/onswipe                   OnSwipe                                 5000000.00
/organization/batanga                   Batanga Media                           5000000.00
/organization/samanage                  Samanage                                5000000.00
/organization/blaze-company             Blaze Public Relations                  5000000.00
/organization/blogfrog                  TapInfluence                            5000000.00
/organization/harvest-exchange          Harvest Exchange                        5000000.00
/organization/shoparoo                  InfoScout                               5000000.00
/organization/futureadvisor             FutureAdvisor                           5000000.00
/organization/trustradius               TrustRadius                             5000000.00
/organization/sindeo                    Sindeo                       

In [97]:
#D2 company received the highest investment in Others and Social, Finance, Analytics, Advertising
d2_others=d2[d2.main_sector == "Others"]
d2_others.groupby(["company_permalink","name"]).agg({"raised_amount_usd" : 'sum'}).sort_values(by="raised_amount_usd",ascending=False)
d2_social=d2[d2.main_sector == "Social, Finance, Analytics, Advertising"]
d2_social.groupby(["company_permalink","name"]).agg({"raised_amount_usd" : 'sum'}).sort_values(by="raised_amount_usd",ascending=False)

raised_amount_usd
company_permalink                                  name                                                   
/organization/celltick-technologies                Celltick Technologies                       37500000.00
/organization/mythings                             myThings                                    34000000.00
/organization/zopa                                 Zopa                                        32900000.00
/organization/imagini                              VisualDNA                                   28550000.00
/organization/marketinvoice                        MarketInvoice                               25553007.00
/organization/sumup                                SumUp                                       24293649.00
/organization/amplience                            Amplience                                   24200000.00
/organization/garlik                               Garlik                                      22350000.00
/organization/sportpursuit                         SportPursuit                                22024487.00
/organization/victor                               Victor                                      21500000.00
/organization/opengamma                            OpenGamma                                   21000000.00
/organization/kantox                               Kantox                                      19646334.00
/organization/troux-technologies                   Troux Technologies                          17850000.00
/organization/neul                                 Neul                                        17800000.00
/organization/livebookings                         Livebookings Holdings Ltd.                  16500000.00
/organization/crowdcube                            Crowdcube                                   15862857.00
/organization/coco-communications                  Coco Communications                         15047666.00
/organization/similarweb                           SimilarWeb                                  15000000.00
/organization/exchange-lab                         The Exchange Lab                            14000000.00
/organization/purplebricks                         PurpleBricks                                13364740.00
/organization/icrossing                            iCrossing                                   13000000.00
/organization/captify                              Captify                                     12597433.00
/organization/borro                                Borro                                       12072758.00
/organization/videoplaza                           Videoplaza                                  12000000.00
/organization/metail                               Metail                                      12000000.00
/organization/inspired-capital-plc                 Inspired Capital plc                        12000000.00
/organization/virtensys                            VirtenSys                                   12000000.00
/organization/seedrs                               Seedrs                                      11703571.00
/organization/psytechnics                          Psytechnics                                 11600000.00
/organization/ocean-outdoor                        Ocean Outdoor                               11597927.00
...                                                                                                    ...
/organization/salaryfinance                        SalaryFinance                                6100000.00
/organization/wonga                                Wonga                                        6000000.00
/organization/brandwatch                           Brandwatch                                   6000000.00
/organization/peoplevox                            Peoplevox                                    6000000.00
/organization/transferwise                         TransferWise                                 6000000.00
/organization/hassle-co

In [98]:
#D3 company received the highest investment in Others and Social, Finance, Analytics, Advertising
d3_others=d3[d3.main_sector == "Others"]
d3_others.groupby(["company_permalink","name"]).agg({"raised_amount_usd" : 'sum'}).sort_values(by="raised_amount_usd",ascending=False)
d3_social=d3[d3.main_sector == "Social, Finance, Analytics, Advertising"]
d3_social.groupby(["company_permalink","name"]).agg({"raised_amount_usd" : 'sum'}).sort_values(by="raised_amount_usd",ascending=False)

raised_amount_usd
company_permalink                                  name                                                             
/organization/manthan-systems                      Manthan Systems                                       50700000.00
/organization/komli-media                          Komli Media                                           28000000.00
/organization/shopclues-com                        ShopClues.com                                         25000000.00
/organization/intarvo                              inTarvo                                               21900000.00
/organization/grameen-financial-services           Grameen Financial Services                            21556050.00
/organization/bankbazaar                           BankBazaar.com                                        19000000.00
/organization/microland                            Microland                                             18300000.00
/organization/eka-software-solutions               Eka Software Solutions                                16000000.00
/organization/qyuki                                Qyuki                                                 15400000.00
/organization/travelguru                           TravelGuru                                            15000000.00
/organization/apnaloan                             ApnaPaisa                                             15000000.00
/organization/financial-information-network-ope... Financial Information Network & Operations Pvt        15000000.00
/organization/shiram-credit                        Shiram Credit                                         15000000.00
/organization/capital-float                        Capital Float                                         13000000.00
/organization/yepme-com                            yepme.com                                             12700000.00
/organization/coverfox-insurance-broking           Coverfox Insurance                                    12000000.00
/organization/funds-india                          FundsIndia                                            11000000.00
/organization/emergent-ventures-india              Emergent Ventures India                               10040000.00
/organization/excellence4u                         Excellence4u                                          10000000.00
/organization/loginext-solutions                   LogiNext Solutions                                    10000000.00
/organization/network18                            Network18                                             10000000.00
/organization/unicommerce                          Unicommerce eSolutions Pvt. Ltd.                      10000000.00
/organization/lendingkart                          Lendingkart                                           10000000.00
/organization/razorpay                             Razorpay                                               9000000.00
/organization/industrybuying                       Industrybuying                                         9000000.00
/organization/vizury                               Vizury                                                 9000000.00
/organization/fusion-microfinance                  Fusion Microfinance                                    8300000.00
/organization/prizm-payment-services               Prizm Payment Services                                 8000000.00
/organization/shubham-housing-development-finan... Shubham Housing Development Finance Company            8000000.00
/organization/spire-technologies                   Spire Technologies                                     8000000.00
/organization/indiaideas                           IndiaIdeas                                             7500000.00
/organization/epiance                              Epiance                                                7500000.00
/organization/crmnext                              CRMNEXT                                                7003500.00
/org

In [46]:
#D1 total number of investments(count)
len(d1.index)

12063

In [47]:
#D2 total number of investments(USD)
len(d2.index)

621

In [48]:
#D3 total number of investments(USD)
len(d3.index)

328

In [49]:
#D1 total number of investments(sum)
d1.raised_amount_usd.sum()

107757097294.0

In [50]:
#D2 total number of investments(sum)
d2.raised_amount_usd.sum()

5379078691.0

In [51]:
#D3 total number of investments(sum)
d3.raised_amount_usd.sum()

2949543602.0

In [99]:
# D1 - USA -no of investment in top sector

d1.groupby("main_sector").agg({ "funding_round_permalink" : 'count', "main_sector" : pd.Series.nunique}).sort_values(by="funding_round_permalink",ascending=False)

funding_round_permalink  main_sector
main_sector                                                                  
Others                                                      2950            1
Social, Finance, Analytics, Advertising                     2714            1
Cleantech / Semiconductors                                  2350            1
News, Search and Messaging                                  1583            1
Health                                                       909            1
Manufacturing                                                799            1
Entertainment                                                591            1
Automotive & Sports                                          167            1

In [67]:
# D1 - USA - Count of investment and sum of investment corresponding to each main_sector
pd.merge(d1, d1.pivot_table(values = 'raised_amount_usd',index = ['main_sector'], aggfunc = {'sum','count'}), how="left", on="main_sector").sort_values(by=["count","raised_amount_usd"],ascending=False)

company_permalink  \
12                         /organization/1stdibs   
394                     /organization/agari-data   
496                         /organization/akorri   
617                        /organization/altoweb   
774                   /organization/appcelerator   
840                        /organization/approva   
852               /organization/apreso-classroom   
910                /organization/arena-solutions   
980                /organization/aryaka-networks   
1023               /organization/astute-networks   
1034           /organization/atesto-technologies   
1107                    /organization/authentium   
1109                     /organization/authentix   
1137           /organization/avamar-technologies   
1156                        /organization/avelle   
1163                 /organization/avere-systems   
1267              /organization/bakbone-software   
1376                     /organization/betabrand   
1387               /organization/betterworks-new   
1410              /organization/bigband-networks   
1411                   /organization/bigcommerce   
1619                       /organization/bluearc   
1634                       /organization/bluefly   
1639             /organization/bluenote-networks   
1733                /organization/branch-metrics   
1744         /organization/breakingpoint-systems   
1810       /organization/broadcast-international   
1826                   /organization/bti-systems   
1955           /organization/cape-clear-software   
1957                       /organization/capella   
...                                          ...   
3151            /organization/digital-air-strike   
6699               /organization/movimento-group   
11941             /organization/zero-motorcycles   
4676               /organization/high-gear-media   
4718                        /organization/hmicro   
3292                    /organization/driverside   
10999             /organization/uqm-technologies   
11385        /organization/voicebox-technologies   
5421                    /organization/jetsmarter   
6522                /organization/mission-motors   
11940             /organization/zero-motorcycles   
31                     /organization/3d-robotics   
1118                   /organization/automile-ab   
1342                         /organization/beepi   
3145                          /organization/digg   
3216                     /organization/ditto-com   
3521                          /organization/emma   
4315                   /organization/gilt-groupe   
5866                     /organization/limos-com   
6404                /organization/metal-networks   
6665                   /organization/mota-motors   
7882   /organization/phillips---temro-industries   
8237                 /organization/project-decor   
8321                   /organization/puppet-labs   
8332                      /organization/purecars   
8333                      /organization/purecars   
8442                     /organization/quick-hit   
8507                      /organization/raidcore   
9284                       /organization/sentera   
9431                 /organization/sight-machine   

                               funding_round_permalink funding_round_type  \
12     /funding-round/292b074d073fdd9c7e9d8f372c3aa5f6            venture   
394    /funding-round/08f7137c4fd52f6e9a4852ee9a765e23            venture   
496    /funding-round/5a9ada593b969a2b814a7f040f22af89            venture   
617    /funding-round/dad4cee2e436772ef678c367b5cddb4d            venture   
774    /funding-round/a3cf130fb3f34ee77aa3ad1c1a730c6a            venture   
840    /funding-round/852095203f5ab885d6f45adc627ceaa0            venture   
852    /funding-round/57f51f32a8b50176fc5f975b5535683b            venture   
910    /funding-round/48ff6b6f15f5d5265bef6d932921c292            venture   
980    /funding-round/4c83f9ac374c8c143fed04ef99c5bd95            venture   
1023   /funding-round/c471ea36374046

In [55]:
# D2 - GBR -no of investment in top sector
d2.groupby("main_sector").agg({ "raised_amount_usd" : 'count', "main_sector" : pd.Series.nunique}).sort_values(by="raised_amount_usd",ascending=False)

raised_amount_usd  main_sector
main_sector                                                            
Others                                                 147            1
Social, Finance, Analytics, Advertising                133            1
Cleantech / Semiconductors                             130            1
News, Search and Messaging                              73            1
Entertainment                                           56            1
Manufacturing                                           42            1
Health                                                  24            1
Automotive & Sports                                     16            1

In [100]:
# D2 - GBR - Count of investment and sum of investment corresponding to each main_sector
pd.merge(d2, d2.pivot_table(values = 'raised_amount_usd',index = ['main_sector'], aggfunc = {'sum','count'}), how="left", on="main_sector").sort_values(by=["count","raised_amount_usd"],ascending=False)

company_permalink  \
64               /organization/azure-solutions   
169                /organization/environmentiq   
247                    /organization/ip-access   
359           /organization/notonthehighstreet   
468                      /organization/sensage   
482      /organization/silverrail-technologies   
544                      /organization/tribold   
332              /organization/myoptique-group   
558                        /organization/ukash   
369                        /organization/ocapo   
109          /organization/celona-technologies   
167                    /organization/enigmatic   
281                         /organization/lyst   
503                        /organization/state   
12                  /organization/acs-clothing   
331              /organization/myoptique-group   
22                      /organization/alfresco   
357                        /organization/njini   
13                  /organization/acs-clothing   
72                    /organization/bathempire   
487            /organization/six-degrees-group   
168                    /organization/enigmatic   
430              /organization/prodigy-finance   
376                  /organization/onefinestay   
73                 /organization/bathrooms-com   
155  /organization/eglue-business-technologies   
161               /organization/electric-cloud   
566                   /organization/velocity-2   
360           /organization/notonthehighstreet   
502                   /organization/srl-global   
..                                         ...   
581                      /organization/wahanda   
296              /organization/medicanimal-com   
598                 /organization/xercise4less   
599                 /organization/xercise4less   
536                     /organization/topivert   
491              /organization/smart-holograms   
317     /organization/molecular-products-group   
11                   /organization/accunostics   
569              /organization/veryan-holdings   
218                   /organization/happy-days   
78                      /organization/biomonde   
570              /organization/veryan-holdings   
233               /organization/image-analysis   
408                  /organization/phase-focus   
217       /organization/greenroad-technologies   
274            /organization/light-blue-optics   
215       /organization/greenroad-technologies   
214       /organization/greenroad-technologies   
275            /organization/light-blue-optics   
506                    /organization/streetcar   
59                     /organization/autoquake   
56                     /organization/autoquake   
216       /organization/greenroad-technologies   
431                     /organization/prodrive   
285                     /organization/made-com   
602                  /organization/yasa-motors   
102                       /organization/carwow   
57                     /organization/autoquake   
58                     /organization/autoquake   
114                 /organization/chargemaster   

                             funding_round_permalink funding_round_type  \
64   /funding-round/adb99f479490e33281ab358ba9da2b29            venture   
169  /funding-round/75dfc6681d2f93780158aff5fd9efcde            venture   
247  /funding-round/4321163564fddf7a26b7235d7c791941            venture   
359  /funding-round/1699ed5949d4d06bce01be9da7d9863f            venture   
468  /funding-round/39314de678e7cdba9b1b7390d23bc394            venture   
482  /funding-round/6816b97b970b7b95395f903ce1dd6d0d            venture   
544  /funding-round/36c8164caf2a5283a766e2a85b9d663c            venture   
332  /funding-round/2c959c6ff6c5958c29388be295288d58            venture   
558  /funding-round/74e9144a6356af4c9a01486f731ce1e8            venture   
369  /funding-round/13a507388b97c4da8f5707dab21d9112            venture   
109  /funding-round/495fda48b04605e30d61bd2cdd73c2a5            venture   
167  /funding-round/9cca5e2b

In [57]:
# D3 - IND  -no of investment in top sector
d3.groupby("main_sector").agg({ "raised_amount_usd" : 'count', "main_sector" : pd.Series.nunique}).sort_values(by="raised_amount_usd",ascending=False)

raised_amount_usd  main_sector
main_sector                                                            
Others                                                 110            1
Social, Finance, Analytics, Advertising                 60            1
News, Search and Messaging                              52            1
Entertainment                                           33            1
Manufacturing                                           21            1
Cleantech / Semiconductors                              20            1
Health                                                  19            1
Automotive & Sports                                     13            1

In [58]:
# D3 - IND - Count of investment and sum of investment corresponding to each main_sector
pd.merge(d3, d3.pivot_table(values = 'raised_amount_usd',index = ['main_sector'], aggfunc = {'sum','count'}), how="left", on="main_sector").sort_values(by=["count","raised_amount_usd"],ascending=False)

company_permalink  \
41                             /organization/caratlane   
52         /organization/classteacher-learning-systems   
58                    /organization/comat-technologies   
59                           /organization/commonfloor   
97                          /organization/firstcry-com   
110                           /organization/girnarsoft   
145                     /organization/ipolicy-networks   
150                     /organization/itzcash-card-ltd   
156                 /organization/k-12-techno-services   
167                             /organization/limeroad   
229                        /organization/pepperfry-com   
256                               /organization/roposo   
286                 /organization/tejas-networks-india   
308                /organization/voylla-retail-pvt-ltd   
322                             /organization/zo-rooms   
89                       /organization/eximsoft-trianz   
99                          /organization/firstcry-com   
128                           /organization/homeshop18   
201                               /organization/myntra   
215    /organization/nxtgen-data-center-cloud-services   
173  /organization/maharana-infrastructure-and-prof...   
124         /organization/hippocampus-learning-centres   
244              /organization/quick-heal-technologies   
240                        /organization/prabhat-fresh   
15                               /organization/ariisto   
96                       /organization/firepro-systems   
138                /organization/india-property-online   
209                            /organization/nest-away   
273                             /organization/snapdeal   
212                        /organization/nobel-hygiene   
..                                                 ...   
129                          /organization/hopscotch-4   
172                              /organization/lybrate   
186                     /organization/medwell-ventures   
199                            /organization/mydentist   
208                           /organization/nephroplus   
71                       /organization/cygnus-medicare   
91                                 /organization/eye-q   
70                       /organization/cura-healthcare   
236                       /organization/portea-medical   
102                         /organization/forus-health   
263                   /organization/sequent-scientific   
198                            /organization/mydentist   
261                  /organization/satnav-technologies   
37                             /organization/bookmycab   
69                       /organization/cura-healthcare   
76                               /organization/dogspot   
101                         /organization/forus-health   
42                              /organization/cardekho   
136      /organization/incredible-technologies-pvt-ltd   
175                /organization/mahindra-first-choice   
176                       /organization/makemytrip-com   
177                       /organization/makemytrip-com   
16                          /organization/ather-energy   
325                        /organization/zoomcar-india   
284                      /organization/taxiforsure-com   
324                        /organization/zoomcar-india   
191                    /organization/mistral-solutions   
290                        /organization/tonbo-imaging   
8                       /organization/ani-technologies   
140                          /organization/indianroots   

                             funding_round_permalink funding_round_type  \
41   /funding-round/ead9f057edb0d007c54f76e5cf5f2375            venture   
52   /funding-round/8be54e916f26b7702504530e6719305f            venture   
58   /funding-round/4d24857fc321f2d54c16573a085b090c            venture   
59   /funding-round/2d3d9f68e6902613d06d2aa77852381c            venture   
97   /funding-round/23b1769412d24510dfc7a56ab1f12e4d            v